node/src/main.rs

```diff
@@ -391,26 +405,26 @@ fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
     // Convert the client into a link resolver
     let link_resolver = Arc::new(LinkResolver::from(ipfs_client));
 
-    // Parse the Ethereum URL
-    let (ethereum_network_name, ethereum_node_url) = parse_ethereum_network_and_node(
-        [ethereum_ipc, ethereum_rpc, ethereum_ws]
-            .into_iter()
-            .filter_map(|x| x.to_owned())
-            .next()
-            .expect("one of --ethereum-ipc, --ethereum-rpc or --ethereum-ws must be provided"),
-    )
-    .expect("failed to parse Ethereum connection string");
-
-    // Set up Ethereum transport
-    let (transport_event_loop, transport) = ethereum_ipc
-        .map(|_| Transport::new_ipc(ethereum_node_url))
-        .or_else(|| ethereum_ws.map(|_| Transport::new_ws(ethereum_node_url)))
-        .or_else(|| ethereum_rpc.map(|_| Transport::new_rpc(ethereum_node_url)))
-        .expect("One of --ethereum-ipc, --ethereum-rpc or --ethereum-ws must be provided");
-
-    // If we drop the event loop the transport will stop working.
-    // For now it's fine to just leak it.
-    std::mem::forget(transport_event_loop);
+    let eth_adapters: Vec<(
+        &str,
+        Arc<graph_datasource_ethereum::EthereumAdapter<graph_datasource_ethereum::Transport>>,
+    )> = ethereum_rpc
+        .map(|arguments| parse_ethereum_networks_and_nodes(arguments, ConnectionType::RPC))
+        .or_else(|| {
```

Comment:
Instead of using `or_else` like this, I think it's best if we combine all adapters for all provided flags into a flat hash map. And fail if for any network name, more than one URL is provided.